--- pessimistic_transaction_patch.orig.cc	2018-07-31 13:04:56.000000000 +0800
+++ pessimistic_transaction_patch.cc	2018-07-31 13:10:46.000000000 +0800
@@ -315,6 +315,8 @@ Status WriteCommittedTxn::CommitBatchInt
 Status WriteCommittedTxn::CommitInternal() {
   // We take the commit-time batch and append the Commit marker.
   // The Memtable will ignore the Commit marker in non-recovery mode
+  WriteOptions write_options = write_options_;
+  write_options.disableWAL = false;
   WriteBatch* working_batch = GetCommitTimeWriteBatch();
   WriteBatchInternal::MarkCommit(working_batch, name_);
 
@@ -326,7 +328,7 @@ Status WriteCommittedTxn::CommitInternal
   // in non recovery mode and simply insert the values
   WriteBatchInternal::Append(working_batch, GetWriteBatch()->GetWriteBatch());
 
-  auto s = db_impl_->WriteImpl(write_options_, working_batch, nullptr, nullptr,
+  auto s = db_impl_->WriteImpl(write_options, working_batch, nullptr, nullptr,
                                log_number_);
   return s;
 }
@@ -359,9 +361,11 @@ Status PessimisticTransaction::Rollback(
 }
 
 Status WriteCommittedTxn::RollbackInternal() {
+  WriteOptions write_options = write_options_;
+  write_options.disableWAL = false;
   WriteBatch rollback_marker;
   WriteBatchInternal::MarkRollback(&rollback_marker, name_);
-  auto s = db_impl_->WriteImpl(write_options_, &rollback_marker);
+  auto s = db_impl_->WriteImpl(write_options, &rollback_marker);
   return s;
 }
 
